python/paddle/fluid/layers/control_flow.py

```diff
@@ -1087,7 +1104,7 @@ def body(i, ten):
                              "(length and structure) as loop_vars: {0}".format(
                                  e))
         now_cond = cond(*output_vars)
-        map_structure(assign, output_vars, loop_vars)
+        assign_skip_lod_tensor_array(output_vars, loop_vars)
```

Comment:
Just a tip, nothing to change:

You are re-implementing `map_structure` in `assign_skip_lod_tensor_array`, now since the PR is ready, you don't have to change. But in the future, you can write something like:

```
def assign_skip_lod_tensor_array(inputs, outputs):
       if inputs.type != core.VarDesc.VarType.LOD_TENSOR_ARRAY:
              assign(inputs, outputs)
```

Then you can just write `map_structure(assign_skip_lod_tensor_array, output_vars, loop_vars)` here. Code change will be less.

Reply:
Thanks, done.